### DISCUSSION TITLE
Converting sophia code 0.6.2 to 0.8

### DISCUSSION THREAD

**Discussion by 1g0rkR** [pchampin/sophia_rs#158] [Q&A] [answered]
Hi folks, I would appreciate for some advises converting code sophia version 0.6.2 to 0.8. 
1. Which type I can use from last version similar to Term v.0.6.2? Is it SimpleTerm, right?
2. What I can use to convert str, String, bool to literal?
I see, there is a small piece of code `g.insert(&jwt, &s_name, &"jwt".as_literal())?;`

**Comment by pchampin** [comment, marked as answer]
> Hi folks, I would appreciate for some advises converting code sophia version 0.6.2 to 0.8.
> 
>  1. Which type I can use from last version similar to Term v.0.6.2? Is it SimpleTerm, right?

SimpleTerm is probably your best bet, indeed -- its closest equivalent in 0.6.2 was [MownTerm](https://docs.rs/sophia_term/0.6.2/sophia_term/type.MownTerm.html).
If you need cheap cloning, you might also want to consider [RcTerm](https://docs.rs/sophia_term/latest/sophia_term/type.RcTerm.html) or [ArcTerm](https://docs.rs/sophia_term/latest/sophia_term/type.ArcTerm.html).

> 2. What I can use to convert str, String, bool to literal?
>     I see, there is a small piece of code `g.insert(&jwt, &s_name, &"jwt".as_literal())?;`

`&str` now implements the [Term] directly, so you can pass it directly, so you can pass "jwt" directly to the insert method, no need for as_literal.
As a consequence, for `String`, you can also pass the result of `as_str()` to insert.
Other primitive types such as i32 and f64 also implement the [Term] trait, but unfortunately, I forgo about `bool` :-(
I created [an issue](https://github.com/pchampin/sophia_rs/issues/159) for that. Any PR welcome, I will included it in a minor version of 0.8 as soon as it is implemented.

In the meantime, you will need to manually convert `bool` to `SimpleTerm`; should look like:

```rust
SimpleTerm::LiteralDatatype(
    (if b { "true" } else { "false" }).into(),
    xsd::bool.iri().unwrap(),
)
```

[Term]: https://docs.rs/sophia_api/0.8.0/sophia_api/term/trait.Term.html

**Comment by 1g0rkR** [comment]
Yeah, thanks. I am converting bool like: `let lit_vs = vs.as_str() * xsd::boolean;`

**Reply by pchampin** [reply]
yes, that's actually even better :-)
